### PR TITLE
feat: add TreeView effect visual test

### PR DESCRIPTION
**Pull request by augustobmoura**
[FX-2844](https://toptal-core.atlassian.net/browse/FX-2844)

### Description

Add happo tests for TreeView

### How to test

- Check the test coverage and the Happo screenshots

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| Insert screenshots or screen recordings | Insert screenshots or screen recordings |

### Development checks

- Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- Annotate all `props` in component with documentation
- Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure that all PR checks are green
- Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

### CODE REVIEW

File: cypress/component/TreeView.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+const component = 'TreeView'
+
+describe('TreeView', () => {
+  it('render with modal', () => {
+    cy.mount(<ModalExample />)
+
+    cy.getByTestId('open').as('open-modal-button').realClick()
+
+    cy.getByTestId('tree-dialog-content').as('dialog-tree').should('be.visible')
```
Comment (ozgurkececioglu) [inline cypress/component/TreeView.spec.tsx:117]: nit: I think `as` usages are redundant

**Comment by ozgurkececioglu** [comment]
I don't know why but the new screenshot for cypress side has a problem. Right borders for the nodes are invisible.

Screen Shot 2022-06-21 at 10 02 43

**Comment by TomasSlama** [comment]
@ozgurkececioglu @augustobmoura , I have played with it in [source](https://happo.io/a/675/snap-requests/8156263/TreeView/with-modal%2Fafter-modal-opened) and it seems we have `box-sizing: inherit`. We need to change that to `border-box`. Than it started to look OK 
image

File: cypress/component/TreeView.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+    cy.getByRole('presentation').happoScreenshot({
+      component,
+      variant: 'with-modal/after-modal-opened',
+    })
+  })
+})
```
Comment (dmaklygin) [inline cypress/component/TreeView.spec.tsx:124]: do we need to test custom `zoom` functionality?

**Review by dmaklygin — APPROVED** [review]

**Review by TomasSlama — APPROVED** [review]
We need to fix the screenshot from cypress (☝️ [comment](https://github.com/toptal/picasso/pull/2892#issuecomment-1161377558)), otherwise it looks good 👍

File: packages/picasso/src/TreeView/story/Horizontal.example.tsx [hunk 3]
```diff
@@ -136,6 +136,13 @@ const createTree = (): TreeNodeInterface => {
       }),
     ],
   })
+
+  return {
+    ...node,
+    selectedOffset: {
+      x: 500,
+    },
+  }
```
Comment (denieler) [inline packages/picasso/src/TreeView/story/Horizontal.example.tsx:145]: it seems this is not necessary code for the example, we need it only for visual tests 🤷‍♂️ I think it's not a good reason to modify code example

File: packages/picasso/src/TreeView/story/CustomZoom.example.tsx [hunk 4]
```diff
@@ -46,6 +46,10 @@ const createTree = (): TreeNodeInterface => {
     info: {
       name: 'NODE+1',
     },
+    selectedOffset: {
+      x: -200,
+      y: 400,
+    },
```
Comment (denieler) [inline packages/picasso/src/TreeView/story/CustomZoom.example.tsx:52]: for example, in this example Zoom buttons are important, not the Tree View itself. But by this change we show the whole Tree View component 🤷‍♂️ Why do our viewers need it?

File: cypress/component/TreeView.spec.tsx (hunk 2 above)
Comment (separatio) [reply cypress/component/TreeView.spec.tsx:124]: We need to make sure that we still support the custom zoom buttons, where needed.

File: cypress/component/TreeView.spec.tsx (hunk 1 above)
Comment (separatio) [reply cypress/component/TreeView.spec.tsx:117]: They are. Removing. :+1:

File: packages/picasso/src/TreeView/story/CustomZoom.example.tsx (hunk 4 above)
Comment (separatio) [reply packages/picasso/src/TreeView/story/CustomZoom.example.tsx:52]: It's not important. Removing.

File: packages/picasso/src/TreeView/story/Horizontal.example.tsx (hunk 3 above)
Comment (separatio) [reply packages/picasso/src/TreeView/story/Horizontal.example.tsx:145]: It looks a lot better, and through this change, we can have the whole tree displayed. Do you think the StaticTreeView horizontal example visual test is enough to cover this case?

**Comment by separatio** [comment]
I've simplified the tests. StaticTreeView tests provide most of the coverage, so we didn't need to modify the Storybook examples. The zoom is good enough, as we're trying to test the zoom buttons, rather than the trees themselves. Thank you @denieler for clarifying this to me.

**Review by denieler — APPROVED** [review]

**Pull request merged**